packages/client/hmi-client/src/components/workflow/ops/model-config/model-config-operation.ts

```diff
@@ -10,6 +10,8 @@ export interface ModelConfigOperationState extends BaseState {
 	transientModelConfig: ModelConfiguration;
 	notebookHistory: NotebookHistory[];
 	hasCodeRun: boolean;
+	datasetModelConfigTaskId?: string;
+	documentModelConfigTaskId?: string;
```

Comment:
Generally speaking, we prefer not to have optional fields as a part of operators' state. It is a lot easier to handle when we know precisely what is there and what is not.

Can we remove the optionals here?